domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgress.kt

```diff
@@ -47,6 +47,22 @@ internal class QuestionAssessmentProgress {
     questionSessionMetrics[currentQuestionIndex].didViewSolution = true
   }
 
+  /**
+   * Called every time the user submits an answer to track how many answers the user submitted per question, along with
+   * any misconceptions.
+   */
+  internal fun answerSubmitted(taggedSkillMisconceptionId: String?) {
```

Comment:
Method names should be present-tense active verbs to describe something that's happening or changing, e.g. ``trackSubmittedAnswer``. The object for which this method is being called is taking some sort of action--we should explicitly call out what that is. The fact that it happens when an answer submitted is a callsite detaill, not an implementation detail for this class. We might, for example, want to track submitted answers at other points in the future not just when the answer is submitted. A clearer name will provide less confusion in that situation.

Reply:
That's a really good point, thanks for pointing it out. Fixed!